hw6/task_hw_6.py

```diff
@@ -0,0 +1,90 @@
+from random import randint as rnd
+
+__all__ = ['common_check', 'print_board', 'all_solutions']
+
+_SIZE = 8
+_COUNT_OPTION = 4
+
+# Добавьте в пакет, созданный на семинаре шахматный модуль.
+# Внутри него напишите код, решающий задачу о 8 ферзях.
+# Известно, что на доске 8×8 можно расставить 8 ферзей так, чтобы они не били друг друга.
+# Вам дана расстановка 8 ферзей на доске, определите, есть ли среди них пара бьющих друг друга.
+# Программа получает на вход восемь пар чисел, каждое число от 1 до 8 - координаты 8 ферзей.
+# Если ферзи не бьют друг друга верните истину, а если бьют - ложь.
+
+
+def _check_dangerous(pos_queen: list) -> bool:
+    size = len(pos_queen)
+    for i, point in enumerate(pos_queen):
+        for j in range(i+1, size):
+            if point[0] == pos_queen[j][0] \
+            or point[1] == pos_queen[j][1] \
+            or abs(point[0] - pos_queen[j][0]) == abs(point[1] - pos_queen[j][1]):
+                return False
```

Comment:
Можно сделать чуть изящней 
```suggestion
    for i, (x0, y0) in enumerate(pos_queen, 1):
        for x1, y1 in pos_queen[i:]:
            if x0 == x1 or y0 == y1 or abs(x0 - x1) == abs(y0 - y1):
                return False
```